### PR TITLE
🏗♻️🚮 Refactor and clean up dev dashboard code

### PR DESCRIPTION
**Pull request by rsimha**
This PR does the following:
- Switches from `babel-plugin-transform-react-jsx` to `@babel/plugin-transform-react-jsx` (part of the `babel` monorepo)
- Changes the negative flag `disable_dev_dashboard_cache` to an opt-in flag `enable_dev_dashboard_cache`
- Makes the output of `gulp serve` less noisy and more pretty
- Moves the dev dashboard tests out of `gulp test` (they were unrelated) and into a new `gulp dev-dashboard` task
- Uses concise logging on Travis for the dev dashboard tests
- Modifies PR check to account for the new task (it doesn't need `gulp build`, and runs really fast, so it belongs in the `Checks` job)

### CODE REVIEW

File: build-system/pr-check/checks.js [hunk 1]
```diff
@@ -72,6 +73,10 @@ function main() {
     if (buildTargets.has('DOCS')) {
       timedExecOrDie('gulp check-links');
     }
+
+    if (buildTargets.has('DEV_DASHBOARD')) {
+      timedExecOrDie('gulp dev_dashboard');
```
Comment (alanorozco) [inline build-system/pr-check/checks.js]: naming nit: `dev_dasboard_tests`

**Comment by alanorozco** [comment]
Thanks for doing this, the new testing command is also easier to remember!

File: build-system/tasks/index.js [hunk 2]
```diff
@@ -25,6 +25,7 @@ require('./compile-expr');
 require('./create-golden-css');
 require('./csvify-size');
 require('./dep-check');
+require('./dev-dashboard');
```
Comment (estherkim) [inline build-system/tasks/index.js]: Missing this file?

File: build-system/app.js [hunk 3]
```diff
@@ -78,9 +78,9 @@ app.get('/serve_mode=:mode', (req, res) => {
 });
 
 if (!global.AMP_TESTING) {
-  if (process.env.DISABLE_DEV_DASHBOARD_CACHE &&
-      process.env.DISABLE_DEV_DASHBOARD_CACHE !== 'false') {
-    devDashboard.setCacheStatus(false);
+  if (process.env.ENABLE_DEV_DASHBOARD_CACHE &&
```
Comment (torch2424) [inline build-system/app.js]: Thank you! 😄

File: build-system/pr-check/checks.js (hunk 1 above)
Comment (torch2424) [reply build-system/pr-check/checks.js]: Agreed.

File: build-system/tasks/serve.js [hunk 4]
```diff
@@ -96,7 +96,7 @@ gulp.task(
         'cache': '  Make local resources cacheable by the browser ' +
             '(default: false)',
         'inspect': '  Run nodemon in `inspect` mode',
-        'disable_dev_dashboard_cache': 'Disables the dev dashboard cache',
+        'enable_dev_dashboard_cache': '  Enables the dev dashboard cache',
```
Comment (torch2424) [inline build-system/tasks/serve.js]: Can we mention this is enabled by default?

File: gulpfile.js [hunk 5]
```diff
@@ -1696,7 +1696,7 @@ gulp.task('default', 'Runs "watch" and then "serve"',
         extensions_from: '  Watches and builds only the extensions from the ' +
             'listed AMP(s).',
         noextensions: '  Watches and builds with no extensions.',
-        disable_dev_dashboard_cache: 'Disables the dev dashboard cache',
+        enable_dev_dashboard_cache: '  Enables the dev dashboard cache',
```
Comment (torch2424) [inline gulpfile.js]: Here as well 😄

File: build-system/pr-check/checks.js (hunk 1 above)
Comment (torch2424) [reply build-system/pr-check/checks.js]: Agreed.

File: build-system/app.js (hunk 3 above)
Comment (rsimha) [reply build-system/app.js]: 👍

File: build-system/pr-check/checks.js (hunk 1 above)
Comment (rsimha) [reply build-system/pr-check/checks.js]: Done.

File: build-system/tasks/index.js (hunk 2 above)
Comment (rsimha) [reply build-system/tasks/index.js]: Yep. Added.

File: build-system/tasks/dev-dashboard.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+const Mocha = require('mocha');
+const {createCtrlcHandler, exitCtrlcHandler} = require('../ctrlcHandler');
+const {isTravisBuild} = require('../travis');
+
+
+/**
+ * Run all the dev dashboard tests
+ */
+function runDevDashboardTests() {
+  const mocha = new Mocha({reporter: isTravisBuild() ? 'dot' : 'spec'});
```
Comment (alanorozco) [inline build-system/tasks/dev-dashboard.js]: 🎉

File: build-system/tasks/serve.js (hunk 4 above)
Comment (rsimha) [reply build-system/tasks/serve.js]: Done.

File: gulpfile.js (hunk 5 above)
Comment (rsimha) [reply gulpfile.js]: Done.

**Comment by rsimha** [comment]
Comments addressed. Will let @estherkim review before merging.

**Pull request merged**